kotlin/android/app/src/main/java/dev/firezone/android/features/session/ui/ResourceViewModel.kt

```diff
@@ -0,0 +1,58 @@
+/* Licensed under Apache 2.0 (C) 2024 Firezone, Inc. */
+package dev.firezone.android.features.session.ui
+
+import dev.firezone.android.core.data.ResourceState
+import dev.firezone.android.core.data.stateSymbol
+import dev.firezone.android.tunnel.model.Resource
+import dev.firezone.android.tunnel.model.ResourceType
+import dev.firezone.android.tunnel.model.Site
+import dev.firezone.android.tunnel.model.StatusEnum
+import dev.firezone.android.tunnel.model.isInternetResource
+
+data class ResourceViewModel(
+    val id: String,
+    val type: ResourceType,
+    val address: String?,
+    val addressDescription: String?,
+    val sites: List<Site>?,
+    val displayName: String,
+    val name: String,
+    val status: StatusEnum,
+    var state: ResourceState,
+)
+
+fun internetResourceDisplayName(
+    resource: Resource,
+    state: ResourceState,
+): String {
+    return "${state.stateSymbol()} ${resource.name}"
+}
+
+fun Resource.toResourceViewModel(resourceState: ResourceState): ResourceViewModel {
```

Comment:
Shouldn't this function be defined on the Resource class?

Reply:
It's defined in the `Resource` class but not on the `Resource` module.